vm/actor/src/builtin/miner/mod.rs

```diff
@@ -820,17 +1244,25 @@ impl Actor {
         rt.validate_immediate_caller_accept_any()?;
         let st: State = rt.state()?;
 
-        let sec = st.get_sector(rt.store(), params.sector_number);
-        if let Ok(None) = sec {
-            ActorError::new(
+        match st.get_sector(rt.store(), params.sector_number) {
+            Err(e) => Err(ActorError::new(
+                ExitCode::ErrIllegalState,
+                format!(
+                    "failed to load proven sector {}: {}",
+                    params.sector_number, e
+                ),
+            )),
+            Ok(None) => Err(ActorError::new(
                 ExitCode::ErrNotFound,
-                format!("sector hasn't been proven {}", params.sector_number),
-            );
+                format!("sector {} not proven", params.sector_number),
+            )),
+            Ok(_sector) => Ok(()),
```

Comment:
Technically more clear if you say Ok(Some(_)) instead of Ok(_)

Reply:
Absolutely!